documentation/functions/domain/IGNORE.md

```diff
@@ -1,8 +1,162 @@
+be modifying the records.
+
+To solve this problem simply include `IGNORE()` statements that identify which records
+are managed elsewhere and should be ignored
+
+Technically `IGNORE_NAME` is a promise that DNSControl will not modify or
+delete existing records that match particular patterns. It is like
+[`NO_PURGE`](../domain/NO_PURGE.md) that matches only specific records.
+
+Including a record that is ignored is considered an error and may have undefined behavior.
+
+## Syntax
+
+```
+IGNORE(labelSpec, typeSpec, targetSpec):
+IGNORE(labelSpec, typeSpec):
+IGNORE(labelSpec):
+```
+
+* `labelSpec` is a glob that matches the DNS label. For example `"foo"` or `"foo*"`.  `"*"` matches all labels, as does the empty string (`""`).
+* `typeSpec` is a comma-separated list of DNS types.  `"*"` matches all DNS types, as does the empty string (`""`).  For example "A" matches DNS A records, "A,CNAME" matches both A and CNAME records.
+* `targetSpec` is a glob that matches the DNS target. For example `"foo"` or `"foo*"`.  `"*"` matches all targets, as does the empty string (`""`).
+
+typeSpec and targetSpec default to `"*"` if they are omitted.
+
+## Globs
+
+The `labelSpec` and `targetSpec` parameters supports glob patterns in the style
+of the [gobwas/glob](https://github.com/gobwas/glob) library.  All of the
+following patterns will work:
+
+* `IGNORE("*.foo")` will ignore all records in the style of `bar.foo`, but will not ignore records using a double
+subdomain, such as `foo.bar.foo`.
+* `IGNORE("**.foo")` will ignore all subdomains of `foo`, including double subdomains.
+* `IGNORE("?oo")` will ignore all records of three symbols ending in `oo`, for example `foo` and `zoo`. It will
+not match `.`
+* `IGNORE("[abc]oo")` will ignore records `aoo`, `boo` and `coo`. `IGNORE("[a-c]oo")` is equivalent.
+* `IGNORE("[!abc]oo")` will ignore all three symbol records ending in `oo`, except for `aoo`, `boo`, `coo`.        `IGNORE("[!a-c]oo")` is equivalent.
+* `IGNORE("{bar,[fz]oo}")` will ignore `bar`, `foo` and `zoo`.
+* `IGNORE("\\*.foo")` will ignore the literal record `*.foo`.
+
+## Examples
+
+General examples:
+
+{% code title="dnsconfig.js" %}
+```javascript
+D("example.com",
+  IGNORE("foo"), // matches any records on foo.example.com
+  IGNORE("baz", "A"), // matches any A records on label baz.example.com
+  IGNORE("*", "MX", "*"), // matches all MX records
+  IGNORE("*", "CNAME", "dev-*"), // matches CNAMEs pointing to hosts that start with dev-*
+  IGNORE("bar", "A,MX"), // ignore only A and MX records for name bar
+  IGNORE("*", "*", "dev-*), // Ignore targets with a `dev-` prefix
+  IGNORE("*", "A", "1\.2\.3\."), // Ignore targets that match the 1.2.3.0/24 CIDR block.
+);
+```
+{% endcode %}
+
+Ignore Let's Encrypt (ACME) validation records:
+
+```
+  IGNORE("_acme-challenge.**", "TXT"),
+```
+
+Ignore DNS records typically inserted by Microsoft ActiveDirectory:
+
+```
+  IGNORE("_gc.**", "SRV"), // General Catalog
```

Comment:
I'm missing the `D()` domain setup and the code highlighting.

```suggestion
{% code title="dnsconfig.js" %}
```javascript
D("example.com",
  IGNORE("_gc.**", "SRV"), // General Catalog
```